### PR TITLE
ICMSLST-1825 - Typeahead name input for constabulary search

### PR DESCRIPTION
**Pull request by chris-pettinga**
Implemented typeahead for constabulary search form using ModelSelect2Widget

### CODE REVIEW

File: web/domains/constabulary/forms.py [hunk 1]
```diff
@@ -31,6 +32,23 @@ class Meta:
         model = Constabulary
         fields: list[Any] = []
 
+    @property
+    def form(self):
+        form = super().form
+        form.fields["name"] = ModelChoiceField(
+            queryset=Constabulary.objects.all(),
```
Comment (MattHolmes123) [inline web/domains/constabulary/forms.py:39]: there is an `is_active` field on the Constabulary model.
Not sure if it is used in practice but we should filter on it.

File: web/templates/web/domains/constabulary/list.html [hunk 2]
```diff
@@ -9,3 +14,8 @@
 {% block empty_message %}
         There aren't any constabularies that match your search
 {% endblock %}
+
+{% block page_js %}
+  {{ super() }}
+  {{ filter.form.media.js }}
+{% endblock %}
```
Comment (MattHolmes123) [inline web/templates/web/domains/constabulary/list.html:21]: missing newling at end of file.

File: web/tests/domains/constabulary/test_forms.py [hunk 3]
```diff
@@ -59,6 +59,13 @@ def test_filter_order(self):
         assert first.name == "Big London Constabulary"
         assert last.name == "That Constabulary"
 
+    def test_form_name_field(self):
+        """Tests that the form field is a ModelChoiceField with the correct ModelSelect2Widget."""
+        form = ConstabulariesFilter().form
+        assert form.fields["name"].widget.__class__.__name__ == "ModelSelect2Widget"
+        assert form.fields["name"].queryset.model == Constabulary
+        assert form.fields["name"].widget.search_fields == ("name__icontains",)
```
Comment (MattHolmes123) [inline web/tests/domains/constabulary/test_forms.py:67]: Do you want to add a test for the filter?

Comment (chris-pettinga) [reply web/tests/domains/constabulary/test_forms.py:67]: We already have one above:

```
    def test_name_filter(self):
        results = self.run_filter({"name": "constab"})
        assert results.count() == 2
```

Comment (MattHolmes123) [reply web/tests/domains/constabulary/test_forms.py:67]: Sorry! I was rushing yesterday and didn't check.

File: web/domains/constabulary/forms.py [hunk 4]
```diff
@@ -31,6 +32,23 @@ class Meta:
         model = Constabulary
         fields: list[Any] = []
 
+    @property
+    def form(self):
+        form = super().form
+        form.fields["name"] = ModelChoiceField(
+            queryset=Constabulary.objects.filter(is_active=True),
```
Comment (MattHolmes123) [inline web/domains/constabulary/forms.py:39]: I was wrong to suggest this change:
`queryset=Constabulary.objects.filter(is_active=True),`
The screen can be used to search for archived Constabularies
my bad

**Review by MattHolmes123 — APPROVED** [review]

**Review by marcuspp — APPROVED** [review]

**Pull request merged**